packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -102,6 +136,129 @@ function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
+      ...(pd.patientMatch ? { patientResource: pd.patientResource } : undefined),
     };
   });
 }
+
+export async function runNextPdOnNewDemographics({
+  patient,
+  facilityId,
+  requestId,
+  cqLinks,
+}: {
+  patient: Patient;
+  facilityId: string;
+  requestId: string;
+  cqLinks: CQLink[];
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const newDemographicsHere = getNewDemographics(updatedPatient, cqLinks);
+  const foundNewDemographicsHere = newDemographicsHere.length > 0;
+  const foundNewDemographicsAcrossHies = await checkLinkDemographicsAcrossHies({
+    patient: updatedPatient,
+    requestId,
+  });
+  if (foundNewDemographicsHere || foundNewDemographicsAcrossHies) {
+    if (foundNewDemographicsHere) {
+      await updatePatientLinkDemographics({
+        requestId,
+        patient: updatedPatient,
+        source: MedicalDataSource.CAREQUALITY,
+        links: newDemographicsHere,
+      });
+    }
+    await discover({
+      patient: updatedPatient,
+      facilityId,
+      rerunPdOnNewDemographics: false,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.patientDiscovery,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        foundNewDemographicsHere,
+        foundNewDemographicsAcrossHies,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function runNexPdIfScheduled({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledPdRequest = getCQData(updatedPatient.data.externalData)?.scheduledPdRequest;
+
+  if (scheduledPdRequest) {
+    await discover({
+      patient: updatedPatient,
+      facilityId: scheduledPdRequest.facilityId,
+      requestId: scheduledPdRequest.requestId,
+      forceEnabled: scheduledPdRequest.forceCarequality,
+      rerunPdOnNewDemographics: scheduledPdRequest.rerunPdOnNewDemographics,
+    });
+    await resetPatientScheduledPatientDiscoveryRequestId({
+      patient: updatedPatient,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.runScheduledPatientDiscovery,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        scheduledPdRequestId: scheduledPdRequest.requestId,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function queryDocsIfScheduled({
+  patient,
+  isFailed = false,
+}: {
+  patient: Pick<Patient, "id" | "cxId">;
+  isFailed?: boolean;
+}): Promise<void> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledDocQueryRequestId = getCQData(
+    updatedPatient.data.externalData
+  )?.scheduledDocQueryRequestId;
+
+  if (scheduledDocQueryRequestId) {
```

Comment:
nit: could return if not set, so we don't have multiple indentation levels - makes it slightly easier to read. :)

Reply:
Done here and the other if/else functions (for both CW and CQ).